### ISSUE TITLE
Nomic-embed-text crashing on 0.17.0-rc.3

### ISSUE THREAD

**Issue by jtmarmon** [TabbyML/tabby#3101]
**Describe the bug**
On the latest RC (0.17.0-rc3) I see the following logs when starting tabby

```
Starting...2024-09-06T21:51:10.063326Z  WARN llama_cpp_server::supervisor: crates/llama-cpp-server/src/supervisor.rs:98: llama-server <embedding> exited with status code -1, args: `Command { std: "/home/ubuntu/dist/tabby_x86_64-manylinux2014-cuda122/llama-server" "-m" "/home/ubuntu/.tabby/models/TabbyML/Nomic-Embed-Text/ggml/model.gguf" "--cont-batching" "--port" "30888" "-np" "1" "--log-disable" "--ctx-size" "4096" "-ngl" "9999" "--embedding" "--ubatch-size" "4096", kill_on_drop: true }`
```

This was fixed when switching to tabby 0.16.1, which indicates an issue introduced after this release.

More info in this slack thread https://tabbyml.slack.com/archives/C05CWLZ0Y85/p1725658217391779

**Information about your GPU**
Nvidia a10g running on Ubuntu

```
$ nvidia-smi
Fri Sep  6 22:25:04 2024
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.183.01             Driver Version: 535.183.01   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A10G                    Off | 00000000:00:1E.0 Off |                    0 |
|  0%   33C    P8              62W / 300W |      0MiB / 23028MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
```

**Comment by wsxiaoys** [comment]
I'm unable to reproduce this issue – the version of llama.cpp has not been incremented from 0.16 to 0.17, and there have been no other changes (the llama-server binary is built solely from llama.cpp).

Could you please try version 0.17.0-rc.3 once more to see if the issue persists? If possible, provide detailed steps for reproduction. Thank you!

**Comment by jtmarmon** [comment]
Just reproduced on a fresh machine. Reproduction steps:

1. Launch a g5.xlarge instance with Ubuntu
2. `sudo apt-get update`
3. `sudo apt install nvidia-cuda-toolkit unzip`
4. `wget https://github.com/TabbyML/tabby/releases/download/v0.17.0-rc.3/tabby_x86_64-manylinux2014-cuda122.zip`
5. `unzip tabby_x86_64-manylinux2014-cuda122.zip`
6. `cd dist/tabby_x86_64-manylinux2014-cuda122/`
7. `chmod +x tabby llama-server`
8. `/tabby serve --model StarCoder-1B --chat-model Qwen2-1.5B-Instruct --device cuda`

The reported error will show up and the server won't start. After this, on the same instance, if you download 0.16.1 instead of 0.17.0-rc.3 and repeat the subsequent steps, the server boots up fine

**Comment by wsxiaoys** [comment]
Thank you for providing this information; I am now able to reproduce the issue.

I've found that even though version r0.17-rc.3 is broken, the nightly build is still functioning: https://github.com/TabbyML/tabby/releases/tag/nightly

However, the differences don't really reveal any clues: https://github.com/TabbyML/tabby/compare/v0.17.0-rc.3...nightly

**Comment by ainub** [comment]
The crash is due to the fact that from nightly 0.17.0-r1 the llama-server is being compiled with avx2 flags, and therefor won't work on avx1 machines.
A quick fix would be to use tabby from rc.3 and llama-server from rc.1. This should temporarily fix the startup issue, until compilation flags are fixed.

**Comment by wsxiaoys** [comment]
@ainub - nope, this issue being discussed in this thread is irrelevant to AVX2. Re-produce environment are machines with avx2 support. Check  https://tabbyml.slack.com/archives/C05CWLZ0Y85/p1725658217391779 for details (logs)

**Comment by ainub** [comment]
@wsxiaoys For me at least, it was again an AVX related issue:
Program received signal SIGILL, Illegal instruction.
0x0000000000626180 in ggml_nbytes ()
=> 0x626180 <ggml_nbytes+128>:	vpmullq 0x30(%rbx),%ymm2,%ymm2


VPMULLQ (ZMM, K, ZMM, ZMM)
Summary: | "Multiply Packed Integers and Store Low Result"
-- | --
Reference: | https://www.felixcloutier.com/x86/PMULLD:PMULLQ.html
Extension: | AVX512EVEX
Category: | AVX512
ISA-Set: | AVX512DQ_512
CPL: | 3
iform: | VPMULLQ_ZMMu64_MASKmskw_ZMMu64_ZMMu64_AVX512
iclass: | VPMULLQ
ASM: | VPMULLQ

**Comment by wsxiaoys** [comment]
oh shoot - you're right, the flag displayed are about compiling environment: https://github.com/ggerganov/llama.cpp/issues/9147


Thanks for identifying the cause - working on a fix now.

**Comment by wsxiaoys** [comment]
Related: https://github.com/ggerganov/llama.cpp/issues/8734

**Comment by wsxiaoys** [comment]
I've confirmed that the fix in [v0.17.0-rc.6](https://github.com/TabbyML/tabby/releases/tag/v0.17.0-rc.6) resolves the issue. 

@jtmarmon, could you please test it and let us know if it works for you? Thanks!

**Comment by jtmarmon** [comment]
Yep, works! Thanks @wsxiaoys :)
